tests/test_optislang/test_optislang_integration.py

```diff
@@ -80,3 +83,51 @@ def test_simple_solve(load_mixing_elbow_param_case_dat):
 
     assert input_parameters[0] == input_parameters2[0]
     assert output_parameters[0] == output_parameters2[0]
+
+
+@pytest.mark.optislang
+@pytest.mark.integration
+def test_generate_read_mesh(mixing_elbow_geometry):
+
+    """
+    Use case 2: This optiSLang integration test performs these steps
+
+    - Launch Fluent in Meshing Mode
+    - Generate mesh with default workflow settings
+    - Read created mesh file
+    - Switch to solution and write case file
+
+    This test queries the following using PyTest:
+    - Session health
+    """
+
+    # Step 1: Setup logging
+    pyfluent.set_log_level("ERROR")
+    pyfluent.enable_logging_to_stdout()
+
+    # Step 2: Launch fluent session in meshing mode
+    session = pyfluent.launch_fluent(
+        meshing_mode=True, precision="double", processor_count=2
+    )
+    assert session.check_health() == "SERVING"
+
+    # Step 3 Generate mesh from geometry with default workflow settings
+    session.meshing.workflow.InitializeWorkflow(WorkflowType="Watertight Geometry")
+    session.meshing.workflow.TaskObject["Import Geometry"].Arguments = dict(
+        FileName=mixing_elbow_geometry
+    )
+    session.meshing.workflow.TaskObject["Import Geometry"].Execute()
+    session.meshing.workflow.TaskObject["Generate the Volume Mesh"].Execute()
+    session.meshing.tui.mesh.check_mesh()
+    session.meshing.tui.file.write_mesh("default_mesh.msh.gz")
+    session.meshing.tui.file.write_mesh("default_mesh.msh.h5")
+
+    # Step 4: use created mesh file - .msh.gz/.msh.h5
+    session.meshing.tui.file.read_mesh("default_mesh.msh.gz", "ok")
+    session.meshing.tui.file.read_mesh("default_mesh.msh.h5", "ok")
+
+    # Step 5: Switch to solution and Write case file
+    session.meshing.tui.switch_to_solution_mode("yes")
+    session.solver.tui.solve.initialize.hyb_initialization()
+    session.solver.tui.file.write_case("default_case.cas.gz")
+    session.solver.tui.file.write_case("default_case.cas.h5")
```

Comment:
@egravenh we're not checking that these two writes have succeeded

Reply:
Added the following:

        assert (Path(temporary_resource_path) / "default_case.cas.gz").exists() == True
        assert (Path(temporary_resource_path) / "default_case.cas.h5").exists() == True